views/ThingiSettingsItem.qml

```diff
@@ -25,17 +27,31 @@ RowLayout
     {
         id: inputField
         text: thingiSettingsItem.value
+        visible: thingiSettingsItem.type == "text"
         Layout.fillWidth: true
         selectByMouse: true
         onEditingFinished: {
             ThingiService.saveSetting(thingiSettingsItem.key, thingiSettingsItem.value)
         }
     }
 
+    ComboBox
+    {
+        id: inputMenu
+        visible: thingiSettingsItem.type == "combobox"
+        Layout.fillWidth: true
+        textRole: "label"
+        currentIndex: ThingiService.getDriverIndex(thingiSettingsItem.value)
```

Comment:
We should try to get `getDriverIndex` removed from here as it's super specific for one setting, whereas this ComboBox is a generic input field (now it doesn't scale to allow other ComboBoxes in the future).

Reply:
I think this could be resolved by wrapping Combobox in a component with extra function to set index by value? Or using a different component. You don't want to use any of the Cure specific components like ExpandablePopup?

Reply:
These are not the same component right? The one in the header shows the currently selected driver, the one here shows the preferred default driver. So their current values are independent.

Reply:
In my opinion they are the same component functionally. They both expect a List[Dict[str, str]] and need to be able to set the currentIndex by value of a particular dictionary element. The wrapper would just be a ComboBox with a function to get the index and would not be restricted to the driver options.

Reply:
They're only the same visually then. Functionally they do something different (set the active driver vs set the preferred default driver).

Using the same visual component is fine, it could have a callback method as prop that can be used to configure the underlying behaviour.